enhancements/network/ovs-dynamic-cpu-pinning.md

```diff
@@ -0,0 +1,116 @@
+---
+title: ovs-dynamic-cpu-pinning
+authors:
+  - "@zeeke"
+reviewers:
+  - "@trozet"
+  - "@MarSik"
+approvers:
+  - TBD
+creation-date: 2023-03-31
+last-updated: 2023-03-31
+tracking-link:
+  - https://issues.redhat.com/browse/CNF-5977
+---
+
+# OVS Dynamic CPU Pinning
+
+## Summary
+
+Allow OVS daemons to run on non reserved CPU if the network workload requires
+cycles and non-reserved CPU are idle.
+
+## Motivation
+
+OVS runs at reserved cpus when an OCP node is configured cpuManagerPolicy to `static`. (TBD doclink)
+Typically the customers want to minimize the amount of reserved cpus and leave
+as much cpus as possible available for the business related workloads.
+
+We have seen in some customer cases that it is possible to disrupt both the node
+and the cluster when the networking load rises, because OVS cannot keep up with 
+the flow and starts removing routes from its tables. Which just makes the situation 
+worse as the amount of upcalls (software based processing when a connection is not 
+present in the OVS tables) increases and increases the cpu load even further.
+
+
+### User Stories
+
+> As an Openshift cluster administrator
+> I want to make the most of the node's CPUs
+> So that the cluster can endure a high network workload.
+
+### Goals
+
+Every cluster node that is configured with static CPU manager policy and
+a low number of reserved CPU does not experience any excessive latency 
+increase while dealing with high network loads.
+
+### Non-Goals
+
+TBD
+
+## Proposal
+
+The main idea to resolve the issue described above is to allow OVS 
+to spread out to other cpus outside the reserved cpu pool. However 
+a care must be taken to not interfere with cpus that are assigned 
+to latency sensitive workloads.
+
+
+### Workflow Description
+
+The feature is triggered by Node Tuning Operator (NTO) when it detects the 
+performance conditions to turn it on. The logic NTO uses to activate this 
+feature is out of this enhancement's scope. 
+The effect of the following steps will be implemented to turn the feature on:
+- NTO's daemonset moves the ovs daemons to a systemd slice different than the default one. 
+  It lets them to spread over CPUs that are assigned to Kubernetes workload.
```

Comment:
The above is correct. Should we mention explicitly the kube slice here or should we stay generic as you did? I don't know if there are any changes with respect to this in rhel 9.2